File: system-test/dotnet/SystemTests/Legacy/Mod08/SmokeTests/System/ShopSmokeTest.cs
using SystemTests.Legacy.Mod08.Base;
using Dsl.Core.Shop;
using Optivem.Testing;
using Xunit;

namespace SystemTests.Legacy.Mod08.SmokeTests.System;

public class ShopSmokeTest : BaseScenarioDslTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldBeAbleToGoToShop(Channel channel)
    {
        await Scenario(channel)
            .When().GoToShop()
            .Then().ShouldSucceed();
    }
}

File: system-test/java/src/test/java/com/optivem/eshop/systemtest/latest/smoke/system/ShopSmokeTest.java
package com.optivem.eshop.systemtest.latest.smoke.system;

import com.optivem.eshop.systemtest.latest.base.BaseScenarioDslTest;
import com.optivem.eshop.dsl.channel.ChannelType;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

class ShopSmokeTest extends BaseScenarioDslTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldBeAbleToGoToShop() {
        scenario.assume().shop().shouldBeRunning();
    }
}

